cli/cfncluster/cfncluster.py

```diff
@@ -406,79 +368,81 @@ def instances(args):
-                                     event.resource_type, event.logical_resource_id,
-                                     event.resource_status_reason))
+                                    (event.get('Timestamp'), event.get('ResourceStatus'),
+                                     event.get('ResourceType'), event.get('LogicalResourceId'),
+                                     event.get('ResourceStatusReason')))
         else:
             sys.stdout.write('\n')
             sys.stdout.flush()
-    except boto.exception.BotoServerError as e:
-        if e.message.endswith("does not exist"):
-            sys.stdout.write('\r')
-            logger.critical(e.message)
+    except ClientError as e:
+        if e.response.get('Error').get('Message').endswith('does not exist'):
+            logger.critical(e.response.get('Error').get('Message'))
             sys.stdout.flush()
-            sys.exit(0)
+            sys.exit(1)
         else:
             raise e
     except KeyboardInterrupt:
         logger.info('\nExiting...')
         sys.exit(0)
 
 def delete(args):
-    saw_update = False
 
     logger.info('Deleting: %s' % args.cluster_name)
     stack = ('cfncluster-' + args.cluster_name)
 
     config = cfnconfig.CfnClusterConfig(args)
-    cfnconn = boto.cloudformation.connect_to_region(config.region,aws_access_key_id=config.aws_access_key_id,
-                                                    aws_secret_access_key=config.aws_secret_access_key)
+
+    cfn = boto3.client('cloudformation', region_name=config.region,
+                       aws_access_key_id=config.aws_access_key_id,
+                       aws_secret_access_key=config.aws_secret_access_key)
+
     try:
-        cfnconn.delete_stack(stack)
+        cfn.delete_stack(StackName=stack)
         if not args.nowait:
             time.sleep(5)
-        status = cfnconn.describe_stacks(stack)[0].stack_status
+        status = cfn.describe_stacks(StackName=stack).get("Stacks")[0].get('StackStatus')
         sys.stdout.write('\rStatus: %s' % status)
         sys.stdout.flush()
         logger.debug('Status: %s' % status)
-        saw_update = True
         if not args.nowait:
             while status == 'DELETE_IN_PROGRESS':
                 time.sleep(5)
-                status = cfnconn.describe_stacks(stack)[0].stack_status
-                events = cfnconn.describe_stack_events(stack)[0]
-                resource_status = ('Status: %s - %s' % (events.logical_resource_id, events.resource_status)).ljust(80)
+                status = cfn.describe_stacks(StackName=stack).get("Stacks")[0].get('StackStatus')
+                events = cfn.describe_stack_events(StackName=stack).get('StackEvents')[0]
+                resource_status = ('Status: %s - %s' % (events.get('LogicalResourceId'), events.get('ResourceStatus'))).ljust(80)
                 sys.stdout.write('\r%s' % resource_status)
                 sys.stdout.flush()
             sys.stdout.write('\rStatus: %s\n' % status)
```

Comment:
(not related to boto3) we miss a `\n` before `Status` to write the status in a new line at the end of the iteration

Reply:
So the reason the message isn't wrapping is because after it deletes, it hits the `ClientError` handler. I've added a `sys.stdout.write('\n')` for the `ClientError` handler.

```
$ cfncluster delete test
Deleting: test
Status: SQS - DELETE_COMPLETE                                                   Stack with id cfncluster-test does not exist
```